llvm/lib/Support/SHA1.cpp

```diff
@@ -307,3 +307,36 @@ std::array<uint8_t, 20> SHA1::hash(ArrayRef<uint8_t> Data) {
   memcpy(Arr.data(), S.data(), S.size());
   return Arr;
 }
+
+std::string SHA1::hashToString(ArrayRef<uint8_t> Data, StringRef Mapping) {
+  constexpr auto BitsPerChar = 6;
+  constexpr auto CombinationPerChar = 1 << BitsPerChar;
+  constexpr auto LoopIterations = 27;
+  assert(Mapping.size() >= CombinationPerChar);
+  std::array<uint8_t, 20> Hash = SHA1::hash(Data);
+  std::string Res;
+  Res.reserve(LoopIterations);
+
+  unsigned Bits = 0;
+  unsigned Elem = 0;
+  while (1) {
+    uint16_t Value = Hash[Elem] << 8;
+    if (LLVM_LIKELY(Elem + 1 < Hash.size()))
+      Value |= Hash[Elem + 1];
```

Comment:
```suggestion
      Value |= Hash[Elem + 1] << 8;
```
also you need to update the `if` condition to do it only when you have consumed all the 8 MSB bit.